src/action.rs

```diff
@@ -22,6 +26,23 @@ pub(crate) enum MessageAction {
         content: String,
         requires_armed: bool,
     },
+    Ban {
+        user_id: Id<UserMarker>,
+        guild_id: Option<Id<GuildMarker>>,
```

Comment:
Why is this `Option<...>` if we can't do anything with this action if it's null? It should be required and we should avoid creating this (or the other native moderation actions) if we don't have a guild ID to act on.

My understanding is that the only case where we'll receive a message without a guild ID is if someone DMs the bot - but we don't register that intent, so we shouldn't ever know about that scenario.